src/components/OnboardingManager.tsx

```diff
@@ -30,14 +33,17 @@ export default function OnboardingManager({
     // out of bounds redirect
     if (
       onboarding.progress < 0 ||
```

Comment:
when would  `onboarding.progress` be less than 0?

Reply:
when the user presses the back button on the role page

Reply:
to the edit: that is different.

`pageProgress` records the progress corresponding to the current page. `onboarding.progress` records the overall progress. For example, they may have filled out up to `Legal Experience`, but want to go back to edit `Basic Information`. In this case, `pageProgress=1`, `onboarding.progress=3`.

The `-1` is simply an indication that something has gone wrong, since the `OnboardingManager` must have access to the `OnboardingContext`. In hindsight, there may be a better way to indicate this (i.e. through tossing an error), but that would require more refactoring that's irrelevant to the goal of this PR. It's also not possible since we only have `OnboardingManager` defined inside an `OnboardingContext`.

Reply:
sg, thanks for the clarification. i'm not sure what the code cleanliness standards are, but if you think it's fine to leave it in, i'm fine with it too.

Reply:
If the source of confusion is the `onboarding.progress < 0` check, I'd say just treat it as a sanity check. It shouldn't be less than 0, and if it is, something went wrong, so redirect. It's harmless as is, and the utility is marginal, but for the sake of my sanity, I believe it's better to have it than not.

Reply:
But can you elaborate why you think this is a readability issue or source of confusion? Intuitively, I naturally think of an error boundary / sanity check and won't pay it too much mind, but I'm not sure if I see it the same way as you.